src/components/curator-actions/fetch-signatories.ts

```diff
@@ -1,11 +1,27 @@
 import { get } from 'svelte/store';
-import { dotApi } from '../../stores';
+import { dotApi, proxies } from '../../stores';
+import type { MultisigInfo } from '../../types/account';
+import { currentBlockchain } from '../app-bar/blockchains';
+
+export async function fetchAllProxies() {
+	if (get(proxies) !== undefined) {
+		return;
+	}
 
-async function fetchCuratorProxyAddress(accountId: string) {
 	const api = get(dotApi);
-	const proxy = await api.query.Proxy.Proxies.getValue(accountId);
 
-	return proxy[0][0].delegate;
+	const fetchedProxies = await api.query.Proxy.Proxies.getEntries();
+
+	const proxyMap = new Map<string, string>();
+
+	fetchedProxies.forEach((proxy) => {
+		// Only consider proxies that have a single entry and that entry is of type 'Any'.
+		if (proxy.value[0].length === 1 && proxy.value[0][0].proxy_type.type === 'Any') {
+			proxyMap.set(proxy.keyArgs[0], proxy.value[0][0].delegate);
+		}
+	});
```

Comment:
```suggestion
	fetchedProxies.forEach(({ keyArgs: [address], value: [value] }) => {
		// Only consider proxies that have a single entry and that entry is of type 'Any'.
		if (value.length === 1 && value[0].proxy_type.type === 'Any') {
			proxyMap.set(address, value[0].delegate);
		}
	});
```
Would this improve readability?